lib/Runtime/Types/PathTypeHandler.cpp

```diff
@@ -1520,7 +1520,11 @@ namespace Js
         ScriptContext* scriptContext = instance->GetScriptContext();
         DynamicType* cachedDynamicType = nullptr;
         BOOL isJsrtExternalType = instance->GetType()->IsJsrtExternal();
-        DynamicType* oldType = isJsrtExternalType ? 0 : instance->GetDynamicType();
+        // Share the type of all JsrtExternal objects which will have 'newPrototype' as their new prototype.
+        // TypeTransitionMap is a map with key as weakReference and to store it, key need to have a valid heap block.
+        // So instead of using some constant, use newPrototype as the key for saving newType for such objects
+        // in TypeTransitionMap.
+        DynamicType* oldType = isJsrtExternalType ? reinterpret_cast<DynamicType*>(newPrototype) : instance->GetDynamicType();
```

Comment:
True. I have already started working on it.
BTW, we always create a new `JsrtExternalType` for a given `JsrtExternalObject`. If I start caching it in the map of `newPrototype`, we will end up having lot of entries in the map for these types. I am experimenting something for `JsrtExternalType` on similar lines as caching in `CreateObjectType()`. I will update this PR with the fix.